src/Cafe/HW/Latte/Core/LatteTextureLoader.h

```diff
@@ -632,6 +632,50 @@ class TextureDecoder_R4_G4_UNORM_toRGBA4444_vk : public TextureDecoder, public S
 	}
 };
 
+class TextureDecoder_R4G4_UNORM_to_R8G8B8A8 : public TextureDecoder, public SingletonClass<TextureDecoder_R4G4_UNORM_to_R8G8B8A8>
+{
+public:
+	sint32 getBytesPerTexel(LatteTextureLoaderCtx* textureLoader) override
+	{
+		return 4;
+	}
+
+	void decode(LatteTextureLoaderCtx* textureLoader, uint8* outputData) override
+	{
+		for (sint32 y = 0; y < textureLoader->height; y += textureLoader->stepY)
+		{
+			sint32 yc = y;
+			for (sint32 x = 0; x < textureLoader->width; x += textureLoader->stepX)
+			{
+				uint8* blockData = LatteTextureLoader_GetInput(textureLoader, x, y);
+				sint32 pixelOffset = (x + yc * textureLoader->width) * 4;
+				uint8 v0 = (*(uint8*)(blockData + 0));
+
+				uint8 red4 = (v0 >> 4) & 0xF;
+				uint8 green4 = (v0 & 0xF);
+
+				red4 = (red4 << 4) | red4;
+				green4 = (green4 << 4) | green4;
+
+				*(uint8*)(outputData + pixelOffset + 0) = red4;
+				*(uint8*)(outputData + pixelOffset + 1) = green4;
+			}
```

Comment:
You should add:
```
*(uint8*)(outputData + pixelOffset + 2) = 0;
*(uint8*)(outputData + pixelOffset + 3) = 0;
```
otherwise the extra channels have undefined values.

Reply:
shouldn't `pixelOffset + 3` be 255?